backend/spec/controllers/validated_enrollment_snapshots_controller_spec.rb

```diff
@@ -0,0 +1,107 @@
+      before do
+        login(user)
+      end
+
+      context "when user created this enrollment" do
+        let(:original_enrollment) { create(:enrollment, :franceconnect, user: user) }
+
+        it { is_expected.to have_http_status(:ok) }
+      end
+
+      context "when user is the delegue_protection_donnees associated to this enrollment" do
+        let(:original_enrollment) { create(:enrollment, :franceconnect, delegue_protection_donnees: build(:team_member, :delegue_protection_donnees, user: user)) }
+
+        it { is_expected.to have_http_status(:ok) }
+      end
+
+      context "when user is the responsable traitement associated to this enrollment" do
+        let(:original_enrollment) { create(:enrollment, :franceconnect, responsable_traitement: build(:team_member, :responsable_traitement, user: user)) }
+
+        it { is_expected.to have_http_status(:ok) }
+      end
+
+      context "when user is a reporter for the enrollment target api" do
+        let(:user) { create(:user, roles: ["api_particulier:reporter"]) }
+        let(:original_enrollment) { create(:enrollment, :api_particulier) }
+
+        it { is_expected.to have_http_status(:ok) }
+      end
+
+      context "when user is a reporter for the enrollment group" do
+        let(:user) { create(:user, roles: ["api_particulier:cnaf:reporter"]) }
+        let(:original_enrollment) { create(:enrollment, :api_particulier) }
+
+        it { is_expected.to have_http_status(:ok) }
+      end
+
+      context "when user is a only reporter for another group" do
+        let(:user) { create(:user, roles: ["api_entreprise:pole_emploi:reporter"]) }
+        let(:original_enrollment) { create(:enrollment, :api_particulier) }
+
+        it { is_expected.to have_http_status(:forbidden) }
+      end
+
+      context "when user is a only reporter for another target api" do
+        let(:user) { create(:user, roles: ["api_entreprise:reporter"]) }
+        let(:original_enrollment) { create(:enrollment, :api_particulier) }
+
+        it { is_expected.to have_http_status(:forbidden) }
+      end
+    end
+  end
+
+  describe "payload" do
+    subject(:show_enrollment_payload) do
+      get :show, params: {
+        enrollment_id: enrollment.id,
+        id: snapshot.id
+      }
+
+      JSON.parse(response.body)
+    end
+
+    let(:enrollment) do
+      original_enrollment.snapshot!
+      original_enrollment.update!(intitule: "new intitule")
+      original_enrollment.team_members.first.update!(given_name: "new given name")
+      original_enrollment
+    end
+
+    let(:snapshot) { enrollment.last_snapshot }
+    let(:original_enrollment) { create(:enrollment, :franceconnect, user: user) }
+    let(:user) { create(:user) }
+
+    before do
+      login(user)
+    end
+
+    it "renders the snapshot version" do
+      expect(show_enrollment_payload["intitule"]).not_to eq("new intitule")
+      expect(show_enrollment_payload["team_members"][0]["given_name"]).not_to eq("new given name")
```

Comment:
ici j'aurais fait des tests positifs pour vérifier qu'on renvoie bien la bonne data plutôt que négatifs pour vérifier qu'on renvoie pas la version actuelle

Reply:
C'est plus verbeux de set le bon nom (la construction est affreuse) et in fine je trouve que ça ne sert pas réellement le propos ici